app/models/course.rb

```diff
@@ -172,6 +174,45 @@ def allocations_report_fields
     ]
   end
 
+  def spreadsheet_columns
+    [
+      ['Academic Year', '2020/21'],
+      ['Requested By (Name)', accrediting_provider&.provider_name],
+      ['Requested by (UKPRN)', ''],
+      ['Partner ITT Provider (Name)', provider.provider_name],
+      ['Partner ITT provider (UKPRN)', provider.organisations.first&.school_nctl_organisation&.urn],
+      ['Allocation Subject', allocation_subjects.join(' | ')],
+      ['Route', program_type],
+      ['Course Aim', qualification],
+      ['Course Level', 'PG'],
+      ['Min. no. of recruits', ''],
+      ['Forecast no. of recruits ', ''],
+      ['3 Year Intention', ''],
+      ['Awarding Institution', ''],
+      ['Other Institution', '']
+    ]
+  end
+
+  # Outputs an allocations XLSX with a randomly generated suffix to `public/`.
+  # The first argument is an optional prefix for the filename to make it
+  # easier to differentiate among ~200 other similar XLSX files. Defaults to
+  # the provider.provider_code.
+  # The filename will have a UUID prefix so that it can be hosted
+  # and linked to from an Azure bucket, but that the names can't be
```

Comment:
are you implying its a public visible bucket?

Reply:
It's set to the `blob` setting which means yes, individual files are accessible using the link, but it's not enumerable / you can't access other files easily.

Reply:
so why make it public but obfuscated?

Reply:
we will distribute publicly available individual deep links but don't want to make it feasible to guess other file links if you don't have it.

Reply:
maybe? https://blogs.msdn.microsoft.com/jpsanders/2017/10/12/easily-create-a-sas-to-download-a-file-from-azure-storage-using-azure-storage-explorer/

Reply:
> maybe? https://blogs.msdn.microsoft.com/jpsanders/2017/10/12/easily-create-a-sas-to-download-a-file-from-azure-storage-using-azure-storage-explorer/

Yes that's definitely an option, although probably more work than above implementation, given we need to do this for 250-1000 files.